static/app/components/accordion/accordion.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useState} from 'react';
+import styled from '@emotion/styled';
+import {motion} from 'framer-motion';
+
+import {AccordionSummaryProps} from 'sentry/components/accordion/accordionSummary';
+import space from 'sentry/styles/space';
+
+interface AccordionProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * The body of the accordion that is shown & hidden
+   */
+  accordionDetails: React.ReactNode;
+  /**
+   * The header of the accordion that is always shown
+   */
+  accordionSummary: (p: AccordionSummaryProps) => React.ReactNode;
+  /**
+   * Whether the accordion is disabled (cannot be opened)
+   */
+  disabled?: boolean;
+}
+
+function Accordion({
+  disabled = false,
+  accordionDetails,
+  accordionSummary,
+}: AccordionProps) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [allowOverflow, setAllowOverflow] = useState(false);
+  const duration = 0.25; // how long it takes the accordion to open
+
+  function showDetails() {
+    if (disabled) {
+      setIsExpanded(false);
+      return;
+    }
+
+    setIsExpanded(!isExpanded);
+
+    // When the accordion is fully expanded, we should allow overflow
+    if (!allowOverflow) {
```

Comment:
What's overflow?

This isn't like a textwrap css thing is it?

Reply:
i think mdn explains it best 😅 https://developer.mozilla.org/en-US/docs/Web/CSS/overflow